src/transformers/models/wav2vec2/modeling_tf_wav2vec2.py

```diff
@@ -1220,9 +1221,11 @@ def call(
 
         mask_time_indices = kwargs.get("mask_time_indices", None)
         if mask_time_indices is not None:  # apply SpecAugment along time axis with given indices
-            hidden_states = tf.tensor_scatter_nd_update(hidden_states, mask_time_indices, self.mask_spec_embed)
+            mask = tf.cast(tf.logical_not(tf.cast(mask_time_indices, tf.bool)), tf.float32)
+            hidden_states = hidden_states * tf.expand_dims(mask, -1)
```

Comment:
I'm a bit confused here - as it's implemented now this would just zero out the masked hidden states no? But we should set the vectors to `self.mask_spec_embed` instead. Think it makes sense to use tf.where here and cast `self.mask_spec_embed` to be of the same size as `hidden_states`, so:

```
hidden_states = tf.where(mask_time_indices, expanded_mask_spec_embed, hidden_states)
```